ui/tests/acceptance/pki/pki-tidy-test.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { click, currentRouteName, fillIn, visit } from '@ember/test-helpers';
+
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { v4 as uuidv4 } from 'uuid';
+
+import authPage from 'vault/tests/pages/auth';
+import logout from 'vault/tests/pages/logout';
+import enablePage from 'vault/tests/pages/settings/mount-secret-backend';
+import { runCommands } from 'vault/tests/helpers/pki/pki-run-commands';
+import { SELECTORS } from 'vault/tests/helpers/pki/page/pki-tidy';
+
+module('Acceptance | pki tidy', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
+    await authPage.login();
+    // Setup PKI engine
+    const mountPath = `pki-workflow-${uuidv4()}`;
+    await enablePage.enable('pki', mountPath);
+    this.mountPath = mountPath;
+    await runCommands([
+      `write ${this.mountPath}/root/generate/internal common_name="Hashicorp Test" name="Hashicorp Test"`,
+    ]);
+    await logout.visit();
+  });
+
+  hooks.afterEach(async function () {
+    await logout.visit();
+    await authPage.login();
+    // Cleanup engine
+    await runCommands([`delete sys/mounts/${this.mountPath}`]);
+    await logout.visit();
+  });
+
+  test('it configures a manual tidy operation and shows its details and tidy states', async function (assert) {
+    await authPage.login(this.pkiAdminToken);
+    await visit(`/vault/secrets/${this.mountPath}/pki/tidy`);
+    await click(SELECTORS.tidyEmptyStateConfigure);
+    assert.dom(SELECTORS.tidyConfigureModal.configureTidyModal).exists('Configure tidy modal exists');
+    assert.dom(SELECTORS.tidyConfigureModal.tidyModalAutoButton).exists('Configure auto tidy button exists');
+    assert
+      .dom(SELECTORS.tidyConfigureModal.tidyModalManualButton)
+      .exists('Configure manual tidy button exists');
+    await click(SELECTORS.tidyConfigureModal.tidyModalManualButton);
+    assert.dom(SELECTORS.tidyForm.tidyFormName('manual')).exists('Manual tidy form exists');
+    await click(SELECTORS.tidyForm.inputByAttr('tidyCertStore'));
+    await fillIn(SELECTORS.tidyForm.tidyPauseDuration, '10');
+    await click(SELECTORS.tidyForm.tidySave);
+    await click(SELECTORS.cancelTidyAction);
+    assert.dom(SELECTORS.cancelTidyModalBackground).exists('Confirm cancel tidy modal exits');
+    await click(SELECTORS.tidyConfigureModal.tidyModalCancelButton);
+    // we can't properly test the background refresh fetching of tidy status in testing
+    this.server.get(`${this.mountPath}/tidy-status`, () => {
```

Comment:
Should this be set on the beforeEach hook so that it's always available to future tests as well?

Reply:
@hellobontempo added an empty state for tidy and in order to configure tidy the acceptance tests go through the empty states and configure based on the test criteria.

Reply:
We skip the polling loop in testing as well, so this is the mocked response after clicking `cancel tidy`